code/datums/datum.dm

```diff
@@ -221,3 +221,16 @@
 		qdel(D)
 	else
 		return returned
+
+/**
+  * Causes a runtime error
+  */
+/datum/proc/AllowDrop()
+	CRASH("Bad op: datum/AllowDrop() called")
+
+/**
+  * Causes a runtime error
+  */
+/datum/proc/drop_location()
+	CRASH("Bad op: datum/drop_location() called")

```

Comment:
Here crash like in area/drop_location()

Reply:
You're still casting it as an atom while it might not be one. 
It will run the same, but it'd be better for code clarity to just `return parent.drop_location()`

Reply:
I think, no, here clear datum, it dont have parent.

Reply:
no, i meant on _component.dm